docs/source/configuration.md

```diff
@@ -81,22 +88,59 @@ Following fields are required (and can be customized) in the [`challenge_config.
+
+  - **max_submissions_per_month**: Positive integer which tells the maximum number of submissions per month to a challenge phase.( Optional, Default value is `100000`)
+
+  - **max_submissions**: A positive integer that decides the maximum number of total submissions that can be made to the challenge phase.( Optional,  Default value is `100000`)
+
+  - **default_submission_meta_attributes**: These are the default metadata attributes that will be displayed for all submissions, the metadata attributes are method_name, method_description, project_url, and publication_url.
+      ```yaml
+    default_submission_meta_attributes:
+      - name: method_name
+        is_visible: True
+      - name: method_description
+        is_visible: True
+      - name: project_url
+        is_visible: True
+      - name: publication_url
+        is_visible: True
+      ```
+  - **submission_meta_attributes**: These are the custom metadata attributes that can be added by participants for their submissions. The custom metadata attributes are TextAttribute, SingleOptionAttribute, MultipleChoiceAttribute, and TrueFalseField.
+    ```yaml
+    submission_meta_attributes:
+      - name: TextAttribute
+        description: Sample
+        type: text
+        required: False
+      - name: SingleOptionAttribute
+        description: Sample
+        type: radio
+        options: ["A", "B", "C"]
+      - name: MultipleChoiceAttribute
+        description: Sample
+        type: checkbox
+        options: ["alpha", "beta", "gamma"]
+      - name: TrueFalseField
+        description: Sample
+        type: boolean
+        required: True
+    ```
+  - **is_restricted_to_select_one_submission**: True/False (indicates whether to restrict user to select only one submission for leaderboard. Default is `False`)
+  - **is_partial_submission_evaluation_enabled**: True/False (indicates whether partial submission evaluation is enabled. Default is `False`)
+  - **allowed_submission_file_types**: This is a list of file types that are allowed for submission( Optional Default is `.json, .zip, .txt, .tsv, .gz, .csv, .h5, .npy`)
 
 - **dataset_splits**:
 
   Dataset splits define the subset of test-set on which the submissions will be evaluated on. Generally, most challenges have two splits:
 
-  1. **test-dev** (Allow participants to make large number of submissions, let them see how they are doing, and let them overfit)
-  2. **test-challenge** (Allow small number of submissions so that they cannot mimic test-set. Use this split to decide the winners for the challenge)
+  1. **train_split** (Allow participants to make large number of submissions, let them see how they are doing, and let them overfit)
+  2. **test_split** (Allow small number of submissions so that they cannot mimic test_set. Use this split to decide the winners for the challenge)
```

Comment:
```suggestion
  2. **test_split** (Allow a small number of submissions so that they cannot mimic test_set. Use this split to decide the winners for the challenge)
```